[Chore] ESLint + Prettier + Husky 설정

seojeongm opened the pull request:
## 📌 Summary
- close #2 

Vite + React 19 + TypeScript 5.8 프로젝트의 코드 품질 도구(ESLint, Prettier, Husky)를 설정했습니다. 
- React: 19.1.0 (최신)
- TypeScript: 5.8.3
- Vite: 7.0.0
- TailwindCSS: v4 (CSS-first 접근 방식)
- ESLint: 9.29.0 (Flat Config 사용)
- Prettier: 3.6.2
- Husky: 9.1.7 (v8에서 업그레이드)
- lint-staged: 16.1.2

## 📝 Tasks

1. ESLint 설정 현대화
- ESLint 9 Flat Config 형식으로 마이그레이션 (eslint.config.mjs)
- TypeScript ESLint 통합 (@typescript-eslint/parser, typescript-eslint@8.34.1)
- React 19 지원 플러그인 구성
  - eslint-plugin-react@7.37.5
  - eslint-plugin-react-hooks@5.2.0
  - eslint-plugin-react-refresh@0.4.20

2. 보안 & 접근성 강화
- eslint-plugin-jsx-a11y@6.10.2: 웹 접근성 자동 검사
  - alt 텍스트 필수화
  - 키보드 접근성 보장
  - ARIA 속성 검증
- eslint-plugin-security@3.0.1: JavaScript 보안 취약점 검사
  - XSS 방지 규칙
  - 안전하지 않은 정규식 감지
  - eval() 사용 금지

3. React 19 성능 최적화 규칙
- react/jsx-no-bind: 불필요한 함수 재생성 방지
- react/no-array-index-key: 배열 인덱스 키 사용 경고
- react/jsx-no-constructed-context-values: Context 값 최적화
- react/no-unstable-nested-components: 중첩 컴포넌트 방지

4. TypeScript 엄격 모드 강화

```
{
  "strict": true,
  "noUnusedLocals": true,
  "noUnusedParameters": true,
  "noImplicitReturns": true,
  "forceConsistentCasingInFileNames": true,
  "noFallthroughCasesInSwitch": true,
  "noUncheckedSideEffectImports": true
}
```

6. TailwindCSS v4 최적화
- Prettier 설정에서 불필요한 tailwindConfig 참조 제거
- prettier-plugin-tailwindcss@0.6.13로 클래스 자동 정렬
- eslint-plugin-tailwindcss@3.18.0으로 유효성 검사
- CSS-first 접근 방식을 위한 @theme 지시문 설정

7. 추가 코드 품질 도구
- @tanstack/mailto:eslint-plugin-query@5.81.2: React Query 베스트 프랙티스
- eslint-plugin-simple-import-sort@12.1.1: Import 자동 정렬
- eslint-plugin-import@2.32.0:
  - 순환 참조 감지
  - 사용하지 않는 exports 검사
  - Import 순서 강제

8. Git Hooks 자동화
- Husky v8 → v9 업그레이드
- Pre-commit hook 설정:
`pnpm exec lint-staged`
- lint-staged 설정:

```
{
  "*.{ts,tsx,js,jsx}": ["eslint --cache --fix", "prettier --write"],
  "*.{css,md,json}": ["prettier --write"]
}
```

9. 개발자 경험 향상
- ESLint 캐싱 활성화로 성능 개선
- VS Code 설정 파일 추가 (.vscode/settings.json)
- 새로운 npm scripts:

```
{
  "lint": "eslint . --cache",
  "lint:fix": "eslint . --cache --fix",
  "format": "prettier --write \"src/**/*.{js,jsx,ts,tsx,css,md}\"",
  "format:check": "prettier --check \"src/**/*.{js,jsx,ts,tsx,css,md}\"",
  "type-check": "tsc --noEmit"
}
```

## ✅ PR Checklist (To Reviewer)
- 모든 ESLint 플러그인이 최신 버전으로 설치됨
- React 19 및 TypeScript 5.8과의 호환성 확인
- TailwindCSS v4 지원 확인
- pnpm run lint 정상 실행
- pnpm run format 정상 실행
- pnpm run type-check 정상 실행
- Git commit 시 pre-commit hook 작동 확인

## 📸 Screenshot

seojeongm committed:
chore: ESLint, Prettier, Husky, lint-staged 초기 설정

minjaehan commented on .eslintrc.cjs:
```diff
@@ -0,0 +1,63 @@
+module.exports = {
+  plugins: ['tailwindcss', '@tanstack/query', 'import'],
+  extends: [
+    'next/core-web-vitals',
```
'next/core-web-vitals' 는 next.js 전용 ESLint 설정이어서 저희가 세팅하고 있는 환경과는 맞지 않을 수도 있다고 하네요!

제가 찾아본 바로는 'plugin:react/recommended', 'plugin:react-hooks/recommended',과 같이 설정하는 것이 일반적이라고 해요!

저번에 논의 사항에서 올려주신 에어비엔비 설정 스타일을 참고하신 걸로 예상되는데, 저희 개발 환경과 맞을지 한번 더 내일 디코에서 논의해보면 좋을 것 같아요! 

정말 고생하셨습니다 :)

seojeongm replied:
자세히 봐주셔서 정말 감사합니다! 말씀 해주신대로 수정해야 할 것 같아요. 디코에서 뵐게요 감사합니다 :)

seojeongm replied:
📝 [업데이트] ESLint 설정 관련 커밋 추가했습니다.
- next/core-web-vitals 제거
- plugin:react/recommended, plugin:react-hooks/recommended,으로 교체

seojeongm committed:
chore: next 전용 eslint 설정 제거 및 react 기반 extends로 수정

maehwasoo committed:
chore: ESLint 설정 개선

maehwasoo committed:
chore: ESLint 플러그인 추가 및 규칙 강화

접근성(a11y)·보안·import 관련 ESLint 플러그인 및 추천 규칙을 추가
타이핑 및 코드 일관성 옵션을 강화
불필요하거나 과도한 엄격한 일부 타입스크립트 옵션은 off 처리
husky 버전 업데이트 및 pre-commit 명령 단순화

maehwasoo committed:
chore: Tailwind 통합 및 Vite 구성 업데이트
